addon/services/router-scroll.js

```diff
@@ -29,6 +80,111 @@ class RouterScroll extends Service {
         x: 0, y: 0
       }
     });
+
+    // https://github.com/ember-app-scheduler/ember-app-scheduler/pull/773
+    setupRouter(this.router);
+
+    addListener(this.router, 'routeWillChange', this._routeWillChange);
+    addListener(this.router, 'routeDidChange', this._routeDidChange);
+  }
+
+  destroy() {
+    reset();
+
+    removeListener(this.router, 'routeWillChange', this._routeWillChange);
+    removeListener(this.router, 'routeDidChange', this._routeDidChange);
+
+    if (requestId) {
+      window.cancelAnimationFrame(requestId);
+    }
+
+    if (callbackRequestId) {
+      window.cancelAnimationFrame(callbackRequestId);
+    }
+
+    super.destroy(...arguments);
+  }
+
+  /**
+   * Updates the scroll position
+   * it will be a single transition
+   * @method updateScrollPosition
+   * @param {transition|transition[]} transition If before Ember 3.6, this will be an array of transitions, otherwise
+   */
+  updateScrollPosition(transition) {
+    const url = get(this, 'currentURL');
+    const hashElement = url ? document.getElementById(url.split('#').pop()) : null;
+
+    if (get(this, 'isFirstLoad')) {
+      this.unsetFirstLoad();
+    }
+
+    let scrollPosition;
+
+    if (url && url.indexOf('#') > -1 && hashElement) {
+      scrollPosition = { x: hashElement.offsetLeft, y: hashElement.offsetTop };
+    } else {
+      scrollPosition = get(this, 'position');
```

Comment:
What if you use this as the initial value (it gets overwritten in the if block if needed)

Reply:
Another good find.  Thank you Chantal!